api/src/config.ts

```diff
@@ -1,9 +1,29 @@
-const getEnvString = (name: string): string => process.env[name] || '';
-const getEnvNumber = (name: string): number => parseInt(getEnvString(name), 10);
-const getEnvBoolean = (name: string): boolean => getEnvString(name) === 'true';
+const getEnvString = (name: string, defaultValue?: string): string => {
+    let value = '';
+    try {
+        if (!(process.env[name]) && (defaultValue === undefined || defaultValue === '')) {
+            throw new Error('Environment variable is undefined');
+        } else if (!(process.env[name])) {
+            value = defaultValue as string;
+        } else {
+            value = process.env[name] as string;
+        }
+    } catch (e) {
+        // tslint:disable-next-line no-console
+        console.error(e);
```

Comment:
you throw on line 5 but then catch it here and just log it so the caller gets `''` back. I think you want to either remove the `catch` or re-throw the same exception after logging it.

Also mabye log something other than just the error so you can easily figure out where it came from. Like

```
console.error("error getting environment variable %s: %s", name, e);
```